### ISSUE TITLE
Possible retraction / Z hop option

### ISSUE THREAD

**Issue by nmw01223** [supermerill/Slic3r#2912]
### Version
2.3.57.12

### Operating system type + version
Win10

### Behaviour

Feature request:

Possible option to assist stringing. I use PETG a lot, it is sticky stuff and strings more than most. Temp and retraction are calibrated as best I can.

With a model that is two vertical pillars (5x5mm about 100mm apart). I notice that after the travel, little strings (vertical at about 30 deg to vert) of filament build on the destination pillar. These are caused by tiny oozing and as the nozzle approaches the dest. pillar, a minute blob is added as it crosses the perimeter. On the next layer above, the tiny blob is placed a bit further out, being the edge of the blob on the layer below. And so on, it produces a thick string line extending up vertically. After a bit, the line, having no strength, will sag a bit, the nozzle miss it and start a new one.

No messing with retraction and temperature seem to totally get rid of this. But - Z hop does. Put in 0.2mm+ of Z hop and the ooze from the nozzle it always to high to deposit. However - the effect of Z hop is to produce fine strings on the depart pillar, probably because the vertical movement starts the string.

Therefore, it seems to me that what would help would be a Z hop option to either move halfway, then raise, or just slope upwards at an angle from the travel start to finish.

One further refinement. The amount of Z lift needed is related to travel distance and speed, because the nozzle oozes at a roughly constant rate, over a short or small distance (<20mm) no Z hop is needed, by the time I get to 100mm (travel speed 100mm/s), between 0.2 and 0.4mm is needed. Therefore it would be desirable if the Z hop distance in this case could be specified as mm list per mm travelled.

**Comment by neophyl** [comment]
Sounds like this already requested feature https://github.com/supermerill/SuperSlicer/issues/927

**Comment by neophyl** [comment]
Yeah but the last post in that thread was someone else who similarly wanted a different 'flight profile' to the original poster.  They too wanted it to drop down at the end.  As a feature request it makes sense to consolidate something into one with configurable options.
That's why I said 'sounds like'.  It seems several people want something very similar in purpose which is why I suggest adding your points to that issue.

**Comment by nmw01223** [comment]
Ok, fair enough, missed that.

**Comment by nmw01223** [comment]
Some details on this now added to #2912, the same extended feature could handle both.
